src/server/room-controller.spec.js

```diff
@@ -158,42 +159,139 @@ describe('room-controller', () => {
+          sinon.assert.calledWith(roomService.getRoomById, 'roomId');
+        });
+
+        it('should call mapRoom with the room returned by the service', () => {
+          sinon.assert.calledWith(clientDataMapper.mapRoom, privateRoom);
+        });
+
+        it('should not call getRoomInvitations', () => {
+          sinon.assert.notCalled(roomService.getRoomInvitations);
+        });
+
+        it('should call pageRenderer with the right parameters', () => {
+          sinon.assert.calledWith(pageRenderer.sendPage, request, {}, PAGE_NAME.room, { roomDetails: privateRoom, invitations: null });
+        });
+      });
+
+      describe('when some random person calls', () => {
+        const request = {
+          params: {
+            roomId: 'roomId'
+          },
+          user: {
+            _id: 'randomGuy'
+          }
+        };
+
+        beforeEach(() => {
+          roomService.isRoomMemberOrOwner = sandbox.stub().resolves(false);
+        });
+
+        it('should throw a forbidden exception', () => {
+          expect(() => sut.handleGetRoomPage(request, res)).rejects.toThrow(Forbidden);
+        });
+      });
+    });
+
+    describe('when the public room exists', () => {
       const request = {
         params: {
           roomId: 'roomId'
+        },
+        user: {
+          _id: 'someGuy'
         }
       };
 
+      const publicRoom = { ...privateRoom, access: ROOM_ACCESS_LEVEL.public };
       beforeEach(async () => {
-        clientDataMapper.mapRoom.resolves(room);
-        roomService.getRoomById.callsFake(roomId => {
-          if (roomId === room.roomId) {
-            return Promise.resolve(room);
-          }
-          return Promise.resolve();
-        });
+        roomService.getRoomById.resolves(publicRoom);
         await sut.handleGetRoomPage(request, {});
       });
 
-      it('should call getRoomById with roomId', () => {
-        sinon.assert.calledWith(roomService.getRoomById, 'roomId');
-      });
-
-      it('should call mapRoom with the room returned by the service', () => {
-        sinon.assert.calledWith(clientDataMapper.mapRoom, room);
+      it('should not check if the room caller is the owner or a member', () => {
+        sinon.assert.notCalled(roomService.isRoomMemberOrOwner);
       });
 
       it('should call pageRenderer with the right parameters', () => {
-        sinon.assert.calledWith(pageRenderer.sendPage, request, {}, PAGE_NAME.room, { roomDetails: room });
+        sinon.assert.calledWith(pageRenderer.sendPage, request, {}, PAGE_NAME.room, { roomDetails: privateRoom, invitations: null });
       });
     });
 
     describe('when the room does not exist', () => {
       it('should throw a not found exception', () => {
-        expect(() => sut.handleGetRoomPage({ params: { roomId: 'abc' } }).rejects.toThrow(NotFound));
+        expect(() => sut.handleGetRoomPagePage({ params: { roomId: 'abc' } }).rejects.toThrow(NotFound));
```

Comment:
PagePage?

Reply:
Yeah... I'm not sure how that happened... Fixed, thanks!